Sam bjs2 74991

SAM19991 opened the pull request.

SAM19991 committed:
74991

SAM19991 committed:
74991

SAM19991 committed:
74991

MarselYsup commented on src/main/java/school/faang/bjs274991/School.java:
```diff
@@ -0,0 +1,24 @@
+    private long finishTime;
+
+    public int getTotalPoints() {
+        int totalPoints = 0;
```
можно в функциональном стиле, так будет более лаконично 

```java
 return team.stream()
               .mapToInt(Student::getPoints)
               .sum();

MarselYsup commented on src/main/java/school/faang/bjs274991/Tournament.java:
```diff
@@ -0,0 +1,26 @@
+public class Tournament {
+
+    public CompletableFuture<School> startTask(School school, Task task) {
+        return CompletableFuture.supplyAsync(() -> {
```
Использование пула потоков по умолчанию (ForkJoinPool.commonPool) не рекомендуется в производственном коде, так как вы не контролируете его поведение. Передавайте свой собственный пул потоков в качестве аргумента.

Передавай вторым аргументом метода supplyAsync свой ExecutorService

MarselYsup commented on src/main/java/school/faang/bjs274991/Tournament.java:
```diff
@@ -0,0 +1,26 @@
+        return CompletableFuture.supplyAsync(() -> {
+            System.out.printf("%s begin %s (%s)%n", school.getName(), task.getName(), Thread.currentThread().getName());
+            try {
+                Thread.sleep(task.getDifficulty() * 1000L);
```
1000L - можно вынести в константу, так будет проще конфигурировать код и понимать, что обозначает данное значение

SAM19991 committed:
Refactored getTotalPoints using functional style;
injected custom thread pool as second argument in supplyAsync;
added constants where appropriate.
